File: day1.py
#!/usr/bin/env python3

with open("input1.txt") as file:
    input = file.readlines()
input = [int(x.strip()) for x in input]

def solve1():
    for i in input:
        for ii in input:
            if i + ii == 2020:
                print(f'{i} * {ii} = {i*ii}')
                return

def solve2():
    for a in input:
        for b in input:
            for c in input:
                if a+b+c == 2020:
                    print(f'{a} * {b} * {c} = {a*b*c}')
                    return

solve1()
solve2()

File: day1.jl
#!/usr/bin/env julia

input = open("./input1.txt") do file
	lines = readlines(file)
	parse.(Int, lines)
end


function solve_1()
	for a in input, b in input
		if a + b == 2020
			println(a*b)
			return
		end
	end
end

function solve_2()
	for a in input, b in input, c in input
		if a + b + c == 2020
			println(a*b*c)
			return
		end
	end
end

solve_1()
solve_2()
